GoogleForCreators/web-stories-wp
# Hotlinking: Perform CORS proxy requirement check

## Context



We only want to run hotlinked resources through our CORS proxy that actually need it.

## Summary



Performs a simple request upon hotlinking media insertion to see whether the 

## Relevant Technical Choices



There is a new permanent `needsProxy` flag on the `Resource` object that indicates the status.

## To-do



- Testing

## User-facing changes



N/A

## Testing Instructions






This PR can be tested by following these steps:

1. Enable CORS proxy experiment
1. Insert images/videos from various sites for hotlinking, for example from Unsplash (supports CORS) as well as https://wp.stories.google/ (does not support it)
2. Verify that only images from https://wp.stories.google/ are proxied, the others aren't

## Reviews

### Does this PR have a security-related impact?



No

### Does this PR change what data or activity we track or use?



No

### Does this PR have a legal-related impact?



NO

## Checklist




---



Fixes #9367
